src/canberra.adb

```diff
@@ -189,6 +189,38 @@ package body Canberra is
       Object.Next_ID := Object.Next_ID + 1;
    end Play;
 
+   procedure Play_File
+     (Object      : in out Context;
+      Filename    : String;
+      Event_Sound : out Sound;
+      Kind        : Role   := Event;
+      Name        : String := "")
+   is
+      Error : API.Error_Code;
+
+      use type API.Error_Code;
+   begin
+      Error := API.Play (Object.Handle, Object.Next_ID,
+        Interfaces.C.To_C ("media.filename"),
+        Interfaces.C.To_C (Filename),
+        Interfaces.C.To_C ("media.role"),
+        Interfaces.C.To_C (case Kind is
+                             when Event => "event",
+                             when Music => "music"),
+        Interfaces.C.To_C ("media.name"),
+        Interfaces.C.To_C (if Name'Length > 0 then Name else Filename),
+        System.Null_Address);
+
+      if Error = API.Error_Not_Found then
+         raise File_Not_Found_Error with Filename;
+      end if;
+
+      Raise_Error_If_No_Success (Error);
+
+      Event_Sound := (Object.Handle, Object.Next_ID);
```

Comment:
Can you change this so that `Play` and `Play_File` call a new procedure `Play_Sound`? This procedure should have an extra parameter that specifies whether the String value should be interpreted as an event ID or a filename.

Reply:
The two `*_Not_Found_Error` exceptions can be renamed to `Not_Found_Error`.

Reply:
You're right. I abused code duplication. I'll make the requested changes.

Reply:
It should be solved now.